x-pack/plugins/lens/public/xy_visualization/xy_expression.tsx

```diff
@@ -332,7 +336,18 @@ export function XYChart({
         }
         legendPosition={legend.position}
         showLegendExtra={false}
-        theme={chartTheme}
+        theme={{
+          ...chartTheme,
+          ...(shouldEnrichThemeWithValueLabels && {
+            barSeriesStyle: {
+              displayValue: {
+                fill: 'white',
```

Comment:
Your suggestion to add a contrasting stroke color to the text would make this much more readable!